legistar/bills.py

```diff
@@ -450,17 +452,30 @@ def text(self, matter_id, latest_version_value=None):
     def legislation_detail_url(self, matter_id):
         gateway_url = self.BASE_WEB_URL + '/gateway.aspx?m=l&id={0}'
 
-        # we want to supress any session level params for this head request
+        # We want to supress any session level params for this head request,
         # since they could lead to an additonal level of redirect.
         #
         # Per
         # http://docs.python-requests.org/en/master/user/advanced/, we
         # have to do this by setting session level params to None
-        legislation_detail_route = self.head(
+        response = self.head(
             gateway_url.format(matter_id),
-            params={k: None for k in self.params}).headers['Location']
-
-        return urljoin(self.BASE_WEB_URL, legislation_detail_route)
+            params={k: None for k in self.params}
+        )
+
+        # If the gateway URL redirects, the matter is publicly viewable. Grab
+        # its detail URL from the response headers.
+        if response.status_code == 302:
+            legislation_detail_route = response.headers['Location']
+            return urljoin(self.BASE_WEB_URL, legislation_detail_route)
+
+        # If the status code is anything but a 200 or 302, something is wrong.
```

Comment:
if the status code is a 200, let's change the status code to 403 https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/403

so like

```
if response.status_code == 200:
    response.status_code = 403
    raise scrapelib.HTTPError(response)
```

i think this should make for less confusing debugging in the future. (why are we raising an http error when we are getting a 200)

Reply:
Assuming that we are still using the retry logic of scrapelib (which I think we are), you'll also want do update the `accept_response` method to not retry if we see this error.

updating to `response.status_code <= 403` should be fine.

Reply:
this would change the behavior to require downstream users to handle the 403. I can see both the benefits and downsides to that. If you want to keep the current behavior and have it return None if the code is 200, I'm okay with that.

Reply:
To be clear, a status code of 200 does _not_ indicate an error – it just means the bill is not publicly viewable.

Reply:
Nvm, I understood this better on re-read. Change made.